hydrabus_framework/core/engine.py

```diff
@@ -0,0 +1,122 @@
+import os
+import pkgutil
+import sys
+from hydrabus_framework.core.logger import Logger
+from hydrabus_framework.core.command.run import run_module
+from hydrabus_framework.core.command.show import show
+from hydrabus_framework.core.command.set_options import set_options
+from hydrabus_framework.core.command.use import use
+from hydrabus_framework.core.command.back import back
+from hydrabus_framework.core.command.quit import hbf_exit
+from hydrabus_framework.core.command.help import hbf_help
+from hydrabus_framework.modules.base import BaseModule, ABaseModule
+from importlib import import_module
+from prompt_toolkit.shortcuts import PromptSession
+from prompt_toolkit.styles import Style
+
+
+class HydraFramework:
+    """
+    Framework core engine
+    """
+    def __init__(self):
+        self.logger = Logger()
+        self.app_path = sys.path[0]
+        self.current_module = BaseModule()
+        self.modules = self._list_modules()
+        self.prompt_style = Style.from_dict({
+            # User input (default text).
+            '': '#ffffff',
+
+            # Prompt.
+            'username': '#3399ff',
+            'pound': '#3399ff',
+            'path': '#ff0000 bold',
+        })
+        self.prompt = [
+            ('class:username', '[hbf] '),
+            ('class:path', ''),
+            ('class:pound', '> '),
+        ]
+        self.command = [
+            {"name": "show", "descr": "modules|options: Displays modules list, or module options", "run": show},
+            {"name": "help", "descr": "Help menu", "run": hbf_help},
+            {"name": "exit", "descr": "Exit the console", "run": hbf_exit},
+            {"name": "use", "descr": "Load a module by name", "run": use},
+            {"name": "run", "descr": "Run the selected module", "run": run_module},
+            {"name": "back", "descr": "Move back from the current context", "run": back},
+            {"name": "set", "descr": "Sets a context-specific variable to a value", "run": set_options}
+        ]
+
+    def update_prompt(self, module_name):
+        """
+        Update the user prompt
+        :param module_name: The current module name
+        """
+        if module_name != "":
+            self.prompt = [
+                ('class:username', '[hbf] '),
+                ('class:path', '({})'.format(module_name)),
+                ('class:pound', '> '),
+            ]
+        else:
+            self.prompt = [
+                ('class:username', '[hbf] '),
+                ('class:path', ''),
+                ('class:pound', '> '),
+            ]
+
+    def _list_modules(self):
+        """
+        Generate modules path and attributes list
+        :return: List of available modules
+        """
+        modules = []
+        module_name = "hbfmodules"
+
+        try:
+            package = import_module(module_name)
+        except ImportError:
+            self.logger.print('Package {} not found...'.format(module_name), "error")
+            sys.exit(1)
```

Comment:
Your framework must no exit if no module are presents. You should handle this case

Reply:
Done [#cc3c5ae](https://github.com/hydrabus-framework/framework/commit/cc3c5aefdf6eb8ff6843dff35b73c6c5f040e005)